mathesar/rpc/databases/base.py

```diff
@@ -53,3 +54,19 @@ def get(*, database_id: int, **kwargs) -> DatabaseInfo:
     with connect(database_id, user) as conn:
         db_info = get_database(conn)
     return DatabaseInfo.from_dict(db_info)
+
+
+@rpc_method(name="databases.delete")
+@http_basic_auth_login_required
+@handle_rpc_exceptions
+def delete(*, database_oid: int, database_id: int, **kwargs) -> None:
+    """
+    Drop a database from the server.
+
+    Args:
+        database_oid: The OID of the database to delete on the database.
+        database_id: The Django id of the database to connect to.
+    """
+    user = kwargs.get(REQUEST_KEY).user
+    with connect(database_id, user) as conn:
+        drop_database(database_oid, conn)
```

Comment:
I've been thinking more about this and we should only be providing `delete` for databases created in the internal server. Only Mathesar admins should be able to perform this.

When the user requests a delete, these conditions should match:
1. The user should be a Mathesar admin
2. The database should be present within the internal db server
3. The collaborator must use a role that owns the database

And then, in a single transaction:
- connect to the database that's about to be deleted
- change ownership to the internal db role that we use in mathesar_django
- disconnect
- connect to the internal db server with the internal db role
- drop the database

Reply:
Nothing needed in this PR, we can do this for beta.